mongodb/src/main/java/org/hibernate/ogm/datastore/mongodb/MongoDBDialect.java

```diff
@@ -753,13 +773,22 @@ else if ( type.getReturnedClass() == ObjectId.class ) {
 		else if ( type instanceof StringAsObjectIdType ) {
 			return StringAsObjectIdGridType.INSTANCE;
 		}
+		else if ( type instanceof BytesAsBsonBinaryType ) {
+			return BytesAsBsonBinaryGridType.INSTANCE;
+		}
+		else if ( type.getReturnedClass() == Long.class && type.getClass() == SerializableToBlobType.class  ) {
+			return LongAsBinaryGridType.INSTANCE;
+		}
+		else if ( type.getReturnedClass() == byte[].class && type.getClass() == MaterializedBlobType.class  ) {
+			return BytesAsBinaryGridType.INSTANCE;
+		}
```

Comment:
In this case I don't think you have to check the returned class, you can just use:
```
		else if ( type instanceof MaterializedBlobType ) {
			return BytesAsBsonBinaryGridType.INSTANCE;
		}
```